x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -211,29 +211,17 @@ private Builder(RoleDescriptor rd, @Nullable FieldPermissionsCache fieldPermissi
         }
 
         public Builder cluster(Set<String> privilegeNames, Iterable<ConfigurableClusterPrivilege> configurableClusterPrivileges) {
+            ClusterPermission.Builder builder = ClusterPermission.builder();
             List<ClusterPermission> clusterPermissions = new ArrayList<>();
             if (privilegeNames.isEmpty() == false) {
-                clusterPermissions.add(new ClusterPermission.SimpleClusterPermission(ClusterPrivilegeResolver.get(privilegeNames)));
+                for (String name : privilegeNames) {
+                    ClusterPrivilegeResolver.resolve(name).buildPermission(builder);
```

Comment:
probably `builder = ClusterPrivilegeResolver.resolve(name).buildPermission(builder);` I know from an implementation standpoint it does not matter but I believe it does given the interface. If we keep this as is, I suggest we change the interface.